docs/eeps/draft-embulk-system-properties-and-embulk-home.md

```diff
@@ -0,0 +1,144 @@
+---
+EEP: unnumbered
+Title: System-wide Configuration by Embulk System Properties and Embulk Home
+Author: dmikurube
+Status: Draft
+Type: Standards Track
+Created: 2023-06-16
+---
+
+Introduction
+=============
+
+Real operations of Embulk often need system-wide configurations, not just per-plugin user configurations. Embulk had "System Config" for that purpose, but it did not cover sufficient use-cases.
+
+The new system-wide configuration mechanism, "Embulk System Properties", has been introduced along with a new concept of the "Embulk Home" directory, through Embulk v0.10. This EEP explains the motivation to migrate for the Embulk System Properties from System Config, and the design of Embulk System Properties and Embulk Home.
+
+Background
+===========
+
+System Config
+--------------
+
+Embulk had had a system-wide configuration mechanism, called "System Config", since its early age. System Config, however, had some limitations. It was not applicable to everything. On the other hand, it was unnecessarily powerful at the expense of its coverage.
+
+System Config was based on `org.embulk.config.ConfigSource`. `ConfigSource` was managed in [Guice](https://github.com/google/guice), a dependency injection container, along with all other global objects. As a result, System Config could be used only after a certain initialization of the Embulk process. This limitation narrowed down the coverage of System Config.
+
+System Config had powerful representation thanks to Jackson-based `ConfigSource`. It was not just string-based configuration, but it had a tree structure like JSON. However, the power of the representation was not utilized. The actual System Config values were just strings because they were often configured through command-line options.
+
+Use-cases of System-wide Configurations
+----------------------------------------
+
+Let's see some actual use-cases of system-wide configurations.
+
+### Logging
+
+Logging-related configurations are a typical use-case of system-wide configurations. Embulk had configurations for the log level and the log file path in System Config.
+
+It, however, restricted logging in Embulk. System Config was available only after some initialization, then, logging was avaiable only after the initialization. Messages before the initialization had to be printed by `System.out` or `System.err` out of the control of the SLF4J logging framework.
```

Comment:
avaiable -> available

Reply:
Fixed it along with b66db914ce7facb66744d3167d76920db178f31b